aten/src/ATen/cuda/CUDAGeneratorImpl.cpp

```diff
@@ -75,13 +75,13 @@ Generator createCUDAGenerator(DeviceIndex device_index) {
  * CUDAGeneratorImpl class implementation
  */
 CUDAGeneratorImpl::CUDAGeneratorImpl(DeviceIndex device_index)
-  : c10::GeneratorImpl{Device(DeviceType::CUDA, device_index),
+  : c10::GeneratorImpl{Device::cuda_unchecked(device_index),
```

Comment:
Hmm, this doesn't seem like it should be unchecked to me

Reply:
I added it because the only call site of that constructor [here](https://github.com/pytorch/pytorch/blob/dea233782539f667d2503f3a9af017f47e508d05/aten/src/ATen/cuda/CUDAGeneratorImpl.cpp#L175) pulls the index from an already-created Device object. I guess the constructor is public though, so it could still be used improperly elsewhere. I'll take it out- this doesn't seem like as much of a hot-path anyway.